src/Dictionaries/IDictionary.h

```diff
@@ -171,25 +171,33 @@ class IDictionary : public IExternalLoadable, public IKeyValueEntity
       * for keys that are not in dictionary. If null pointer is passed,
       * then default attribute value must be used.
       */
+    using RefDefault = std::reference_wrapper<const ColumnPtr>;
+    using RefFilter = std::reference_wrapper<IColumn::Filter>;
+    using DefaultOrFilter = std::variant<RefDefault, RefFilter>;
     virtual ColumnPtr getColumn(
-        const std::string & attribute_name,
-        const DataTypePtr & result_type,
-        const Columns & key_columns,
-        const DataTypes & key_types,
-        const ColumnPtr & default_values_column) const = 0;
+        const std::string & attribute_name [[maybe_unused]],
+        const DataTypePtr & attribute_type [[maybe_unused]],
+        const Columns & key_columns [[maybe_unused]],
+        const DataTypes & key_types [[maybe_unused]],
+        DefaultOrFilter defaultOrFilter [[maybe_unused]]) const = 0;
 
     /** Get multiple columns from dictionary.
       *
       * Default implementation just calls getColumn multiple times.
       * Subclasses can provide custom more efficient implementation.
       */
+    using RefDefaults = std::reference_wrapper<const Columns>;
+    using DefaultsOrFilter = std::variant<RefDefaults, RefFilter>;
     virtual Columns getColumns(
         const Strings & attribute_names,
-        const DataTypes & result_types,
+        const DataTypes & attribute_types,
         const Columns & key_columns,
         const DataTypes & key_types,
-        const Columns & default_values_columns) const
+        DefaultsOrFilter defaultsOrFilter) const
```

Comment:
`defaultsOrFilter` -> `defaults_or_filter`

Reply:
good catch, now I changed them all over and added comment.